dbt/adapters/redshift/connections.py

```diff
@@ -234,10 +234,15 @@ def _iam_user_kwargs(self) -> Dict[str, Any]:
     def _iam_role_kwargs(self) -> Dict[str, Optional[Any]]:
         logger.debug("Connecting to redshift with 'iam_role' credentials method")
         kwargs = self._iam_kwargs
-        kwargs.update(
-            group_federation=True,
-            db_user=None,
-        )
+
+        # It's a role, we're ignoring the user
+        kwargs.update(db_user=None)
+
+        # Serverless shouldn't get group_federation, Provisoned clusters should
+        if "serverless" in self.credentials.host:
```

Comment:
It's still kind of wild that this is the right way to do this.

Reply:
IKR